src/Raven.Server/Utils/DeleteOnCloseZipArchive.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.IO;
+using System.IO.Compression;
+
+namespace Raven.Server.Utils
+{
+    public class DeleteOnCloseZipArchive : ZipArchive
+    {
+        private readonly FileStream _stream;
+
+        public DeleteOnCloseZipArchive(Stream stream) : base(stream)
+        {
+            _stream = stream as FileStream;
+        }
+
+        public DeleteOnCloseZipArchive(Stream stream, ZipArchiveMode mode) : base(stream, mode)
+        {
+            _stream = stream as FileStream;
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            try
+            {
+                base.Dispose(disposing);
+            }
+            finally
+            {
+                if (_stream != null)
+                    PosixFile.DeleteOnClose(_stream.Name);
```

Comment:
Are we guaranteed that the file name will be valid *after* we close the file?

Reply:
Not sure, but better be safe than sorry. Will fix it.

Reply:
Done